packages/iris-grid/src/IrisGridRenderer.ts

```diff
@@ -187,12 +187,14 @@ class IrisGridRenderer extends GridRenderer {
       return;
     }
 
-    const lastGroupedColumn = groupedColumns[groupedColumns.length - 1];
-    const modelIndex = columns.findIndex(
-      c => c.name === lastGroupedColumn.name
-    );
-    const columnX = allColumnXs.get(modelIndex);
-    const columnWidth = allColumnWidths.get(modelIndex);
+    // There seem to be duplicate columns with the same name, so determine the
+    // count of distinct names.
+    const numUniqueNames = new Set(groupedColumns.map(({ name }) => name)).size;
```

Comment:
I would wait for the fix from the JS-API making the de-dupe check unnecessary.

Reply:
@dsmmcken the JS-API has been fixed. I have updated my PR accordingly. I should be a little more resilient for any future changes since it doesn't assume any particular order of the grouped columns now.